### PR TITLE
chore: prettier caching

### PR DESCRIPTION
**Pull request by jaylinski**
### Problem

Prettier formatting and linting can get pretty slow, especially when a SvelteKit project gets bigger and there is transpilation/compilation (eg TypeScript) involved.

### Solution

Prettier v2.7 introduced a `--cache` CLI-option, which can "dramatically improve CLI performance":

* https://prettier.io/blog/2022/06/14/2.7.0.html
* https://prettier.io/docs/en/cli.html#--cache

Since prettier was updated to v2.8 in https://github.com/sveltejs/kit/commit/581c43e1884d7d79e4ae6d41ed8f77a1e0b1732f, this change should just work.

> I also added the `--cache`-flag to "internal" CI checks as a separate commit.

### Discussion

* Caching might not be desired when running prettier in a CI environment, since caches are generally not persisted in CI
* It is possible to set a different `--cache-strategy`, which may makes sense for some use-cases (https://prettier.io/docs/en/cli.html#--cache-strategy)

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: packages/create-svelte/shared/+eslint+prettier/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
 		"eslint-config-prettier": "^8.5.0"
 	},
 	"scripts": {
-		"lint": "prettier --plugin-search-dir . --check . && eslint .",
-		"format": "prettier --plugin-search-dir . --write ."
+		"lint": "prettier --cache --check-search-dir . --check . && eslint .",
```
Comment (benmccann) [inline packages/create-svelte/shared/+eslint+prettier/package.json:6]: why was this changed to `--check-search-dir`? I don't see that in the docs: https://prettier.io/docs/en/plugins.html

Comment (jaylinski) [reply packages/create-svelte/shared/+eslint+prettier/package.json]: That was a search-replace error on my side :see_no_evil: . It's fixed now.

**Review by benmccann — APPROVED** [review]
